### PR TITLE
Changes to the buffered reader

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/io.toit [hunk 1]
```diff
@@ -544,55 +514,66 @@ class BufferedReader implements Reader:
...
     delimiter-pos := index-of '\n'
-    if delimiter-pos == null:
-      rest-size := buffered
+    if delimiter-pos == -1:
+      rest-size := buffered-size
       if rest-size == 0: return null
       return read-string rest-size
 
-    if keep-newlines: return read-string delimiter-pos
+    if keep-newline: return read-string delimiter-pos
```
Comment (floitsch) [inline lib/io.toit:545]: There is a bug here (off-by-one). Fixed in the next PR.

File: lib/io.toit [hunk 2]
```diff
@@ -544,55 +514,66 @@ class BufferedReader implements Reader:
...
   /**
   Reads a line as a string.
 
-  Lines are terminated by a newline character (`'\n'`) except for the final
-    line.
-  Carriage returns (`'\r'`) are removed from lines terminated by `'\r\n'`.
+  If $keep-newline is true, the returned string includes the newline character.
+  If $keep-newline is false, trims the trailing '\r\n' or '\n'. This method
+    removes a '\r' even if the platform is not Windows. If the '\r' needs to be
+    preserved, set $keep-newline to true and remove the trailing '\n' manually.
```
Comment (floitsch) [inline lib/io.toit:534]: The next PR has a bit more documentation what happens if the last line doesn't end with a new line.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/io.toit [hunk 3]
```diff
@@ -544,55 +514,66 @@ class BufferedReader implements Reader:
...
 
   The read bytes and the delimiter are consumed.
+  The returned string does not include the delimiter.
 
   # Errors
   The $delimiter must be available.
   */
-  read-until delimiter -> string:
-    length := index-of delimiter
+  read-string-up-to delimiter/int -> string:
```
[the remark this answers is not in the document]
Comment (floitsch) [reply lib/io.toit:575]: added it to my TODO list, in case it isn't added in a later PR.

**Pull request merged**